FileIO/XmlIO/Qt/XmlGspInterface.cpp

```diff
@@ -39,9 +38,15 @@
 
 namespace FileIO
 {
-XmlGspInterface::XmlGspInterface(ProjectData& project) :
-	XMLInterface(), XMLQtInterface(BaseLib::FileFinder().getPath("OpenGeoSysProject.xsd")),
-	_project(project)
+XmlGspInterface::XmlGspInterface(
+    GeoLib::GEOObjects& geoObjects,
+    std::vector<MeshLib::Mesh*> const& mesh_vector,
+    std::function<void(MeshLib::Mesh* const)> add_mesh_callback)
```

Comment:
@rinkk, I can understand your concerns. But if we keep all parts of `ProjectData` in here, soon `FileIO` will also depend on `ProcessLib` (for BCs). Probably that shouldn't be.
Maybe a different approach would be to move some of the parsing code to the DE directory. Do you think, that's feasible?
E.g., the following case checking code probably should be done in some DE class.
```
if (file_node.compare("geo") == 0)
```
Maybe at some point one could even use the constructor that is provided with `ProjectData` to read the `*.prj` (if we remove `std::abort()` :smile:).
That's at least what I can say after a discussion with @endJunction.